lib/google/gax/api_callable.rb

```diff
@@ -372,10 +380,17 @@ def retryable(a_func, retry_options, kwargs)
     # @param a_func [Proc] a proc to be updated
     # @param timeout [Numeric] to be added to the original proc as it
     #   final positional arg.
+    # @param metadata [Hash] request metadata headers
     # @return [Proc] the original proc updated to the timeout arg
-    def add_timeout_arg(a_func, timeout, kwargs)
-      proc do |request|
-        a_func.call(request, deadline: Time.now + timeout, metadata: kwargs)
+    def add_timeout_arg(a_func, timeout, metadata)
+      proc do |request, block|
+        op = a_func.call(request,
+                         deadline: Time.now + timeout,
+                         metadata: metadata,
+                         return_top: true)
```

Comment:
No idea. I'd be surprised if it made a difference. I can benchmark it using the sample I created Do we have any benchmarks already written?

Reply:
@pongad has worked on Spanner benchmarks (not available in open source), which we may be able to re-use. Otherwise, just spamming a high-throughput API should be good enough. (`batch_annotate_images` is probably not a good choice because of the high expected server latency. The Logging API is probably a good test target.)

Reply:
I looked into this a bit more and the library code that's invoked is the same in both cases, so there should not be any difference.

Reply:
The original comment (spelling) still needs to be addressed.